src/lua/httpc.c

```diff
@@ -175,11 +175,16 @@ luaT_httpc_request(lua_State *L)
 	if (!lua_isnil(L, -1)) {
 		lua_pushnil(L);
 		while (lua_next(L, -2) != 0) {
-			if (httpc_set_header(req, "%s: %s",
-					     lua_tostring(L, -2),
-					     lua_tostring(L, -1)) < 0) {
-				httpc_request_delete(req);
-				return luaT_error(L);
+			if (lua_strlen(L, -2) + lua_strlen(L,-1) <= TT_STATIC_BUF_LEN*8) {
```

Comment:
This check can be removed if you'll check a vsnprintf() return value above.

Reply:
Remove this check and add new check in httpc.c